libglusterfs/src/inode.c

```diff
@@ -562,12 +566,14 @@ __inode_ref(inode_t *inode, bool is_invalidate)
             inode->in_invalidate_list = false;
             inode->table->invalidate_size--;
         } else {
+            GF_ASSERT(inode->in_lru_list);
             inode->table->lru_size--;
         }
         if (is_invalidate) {
             inode->in_invalidate_list = true;
             inode->table->invalidate_size++;
             list_move_tail(&inode->list, &inode->table->invalidate);
+            inode->in_lru_list = _gf_false;
```

Comment:
This shouldn't be done here. We should do this when `lru_size` is modified.

Reply:
This comment is still not addressed. Why is it important ?

Reply:
done